src/mito2/src/sst/parquet/row_group.rs

```diff
@@ -267,6 +328,22 @@ impl<'a> RowGroups for InMemoryRowGroup<'a> {
     }
 }
 
+//  Port from https://github.com/apache/arrow-rs/blob/802ed428f87051fdca31180430ddb0ecb2f60e8b/object_store/src/util.rs#L74-L83
+/// Takes a function and spawns it to a tokio blocking pool if available
+pub async fn maybe_spawn_blocking<F, T>(f: F) -> Result<T>
+where
+    F: FnOnce() -> Result<T> + Send + 'static,
+    T: Send + 'static,
+{
+    match tokio::runtime::Handle::try_current() {
+        Ok(runtime) => runtime
+            .spawn_blocking(f)
+            .await
+            .map_err(|e| ParquetError::External(Box::new(e)))?,
+        Err(_) => f(),
+    }
```

Comment:
I'm not sure, should we use `common_runtime::spawn_blocking_read`?

Reply:
Maybe we should remove the read runtime (and other runtimes) in the future. @QuenKar I'm not sure whether spawning it to the read runtime affects performance.